File: packages/modules/api/python/src/api/football_data.py
import httpx

BASE_URL = "https://api.football-data.org/v4"


class FootballDataClient:
    def __init__(self, auth_token: str) -> None:
        self._auth_token = auth_token
        self._client = httpx.AsyncClient()
        self._headers = {"X-Auth-Token": auth_token}

    async def get_competitions(self) -> dict:
        response = await self._client.get(
            f"{BASE_URL}/competitions", headers=self._headers
        )
        return response.json()

    async def get_competition(self, id: int) -> dict:
        response = await self._client.get(
            f"{BASE_URL}/competitions/{id}", headers=self._headers
        )
        return response.json()

    async def get_teams(
        self, limit: int = 50, offset: int = 0
    ) -> dict:
        params = {"limit": limit, "offset": offset}
        response = await self._client.get(
            f"{BASE_URL}/teams", params=params, headers=self._headers
        )
        return response.json()

    async def get_teams_by_competition(self, id: int) -> dict:
        response = await self._client.get(
            f"{BASE_URL}/competitions/{id}/teams", headers=self._headers
        )
        return response.json()

    async def get_team(self, id: int) -> dict:
        response = await self._client.get(
            f"{BASE_URL}/teams/{id}", headers=self._headers
        )
        return response.json()

    async def get_matches_by_team(self, id: int) -> dict:
        response = await self._client.get(
            f"{BASE_URL}/teams/{id}/matches", headers=self._headers
        )
        return response.json()

File: packages/modules/api/go/src/api/football-data.org/football-data.go
package footballdata

import (
	"encoding/json"
	"fmt"
	"net/http"
)

type Competition struct {
	ID          int    `json:"id"`
	Name        string `json:"name"`
	Code        string `json:"code"`
	Type        string `json:"type"`
	Emblem      string `json:"emblem"`
	Plan        string `json:"plan"`
	LastUpdated string `json:"lastUpdated"`
}

type Team struct {
	ID          int    `json:"id"`
	Name        string `json:"name"`
	ShortName   string `json:"shortName"`
	TLA         string `json:"tla"`
	Crest       string `json:"crest"`
	Address     string `json:"address"`
	Website     string `json:"website"`
	Founded     int    `json:"founded"`
	ClubColors  string `json:"clubColors"`
	Venue       string `json:"venue"`
	LastUpdated string `json:"lastUpdated"`
}

type FootballDataClient struct {
	BaseURL   string
	AuthToken string
}

func NewFootballDataClient(authToken string) *FootballDataClient {
	return &FootballDataClient{
		BaseURL:   "https://api.football-data.org/v4",
		AuthToken: authToken,
	}
}

func (c *FootballDataClient) doRequest(path string) (*http.Response, error) {
	req, err := http.NewRequest("GET", c.BaseURL+path, nil)
	if err != nil {
		return nil, err
	}
	req.Header.Set("X-Auth-Token", c.AuthToken)
	return http.DefaultClient.Do(req)
}

func (c *FootballDataClient) GetCompetitions() (map[string]interface{}, error) {
	resp, err := c.doRequest("/competitions")
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

	if resp.StatusCode != http.StatusOK {
		return nil, fmt.Errorf("football-data api returned status: %d", resp.StatusCode)
	}

	var result map[string]interface{}
	if err := json.NewDecoder(resp.Body).Decode(&result); err != nil {
		return nil, err
	}
	return result, nil
}
